chingsley/ireporter
# Ft fetch getallrecords #163058673

#### What does this PR do?
This pull request deals with implementation of fetch to GET ALL records for a given user, thus setting up and populating the record history page for a particular user.

#### Description of Task to be completed?
- create the reporthistory.js file to implement the fetch
- using fetch consume the API with the route /api/v1/red-flags and /api/v1/interventions simultaneously, calling using the 'GET' method
- Display error messages by calling the 'showDialogMsg()' function with the right arguments
- on successful fetch, display a user records in a card-like format.
- If there are no records for the user, display a properly formatted message guiding the user on how to start reporting cases.

#### How should this be manually tested?
- clone the repo
- cd into the root directory
- run `npm install' to install the necessary dependencies
- run `npm test' to see the tests pass
- run `npm start' to start the server
- at the frontend, test the functionalities by trying to sign-up as a new user. Then using your email and password, head over to the login page and sign in. This action will take you to the reporthistory page which handles the fetch to 'GET' all records.

#### What are the relevant pivotal tracker stories
#163058673
